app/scripts/controllers/transactions/index.js

```diff
@@ -430,13 +431,22 @@ export default class TransactionController extends EventEmitter {
     }
 
     if (eip1559Compatibility) {
-      // If the dapp has suggested a gas price, but no maxFeePerGas or maxPriorityFeePerGas
-      //  then we set maxFeePerGas and maxPriorityFeePerGas to the suggested gasPrice.
-      if (
+      if (process.env.EIP_1559_V2 && Boolean(advancedGasFeeDefaultValues)) {
+        txMeta.userFeeLevel = CUSTOM_GAS_ESTIMATE;
+        txMeta.txParams.maxFeePerGas = decGWEIToHexWEI(
+          advancedGasFeeDefaultValues.maxBaseFeeGWEI ||
+            advancedGasFeeDefaultValues.maxBaseFeeMultiplier,
+        );
```

Comment:
Here `decGWEIToHexWEI` for `advancedGasFeeDefaultValues.maxBaseFeeMultiplier` can give wrong values I think.

Reply:
Yes but what will this be a valid value : `decGWEIToHexWEI(advancedGasFeeDefaultValues.maxBaseFeeMultiplier)`.